internal/querycoord/group_balance.go

```diff
@@ -34,8 +35,17 @@ func (b *replicaBalancer) addNode(nodeID int64) ([]*balancePlan, error) {
 		if len(replicas) == 0 {
 			continue
 		}
+
+		offlineNodesCnt := make(map[UniqueID]int, len(replicas))
+		for _, replica := range replicas {
+			for _, nodeID := range replica.NodeIds {
+				if isOnline, err := b.cluster.isOnline(nodeID); err != nil || !isOnline {
+					offlineNodesCnt[replica.ReplicaID]++
+				}
+			}
+		}
 		sort.Slice(replicas, func(i, j int) bool {
-			return len(replicas[i].GetNodeIds()) < len(replicas[j].GetNodeIds())
+			return offlineNodesCnt[replicas[i].ReplicaID] > offlineNodesCnt[replicas[j].ReplicaID]
 		})
```

Comment:
If all replica nodes are online(offlineCnt=0), the new node will be added randomly. Can we add it to the group with least available memory?

Reply:
OK, now this sorts them with the number of offline nodes as first key and the available memory as the second key